### PR TITLE
Create and document build process

### PR DESCRIPTION
**Pull request by txels**
Should include:
- Build script
- Dependency management (bower - installs front-endy things, npm - install js libraries & scripts, pip)
- A minimal documentation on how to run and test

### CODE REVIEW

**Comment by txels** [comment]
This documents the dependency management and test execution, and fixes failing frontend tests.

Merge only once you manage to run all 3 test suites successfully.

**Comment by txels** [comment]
I have created some additional fun scripts for working with this github repo's issues.

To use them you will need to `pip install httpie` and install `hub`. Although it should be easy to implement the pull request with `httpie` directly http://developer.github.com/v3/pulls/#create-a-pull-request

**Comment by txels** [comment]
There, done, removed the `hub` dependency. Now all is piply self contained

**Comment by alexcouper** [comment]
I'm going to be busy all weekend, so I'll hopefully look at this on Monday.

Unrelated: You seen that pycon is streaming online? http://www.timvideos.us/

**Comment by alexcouper** [comment]
I've bailed from the startup weekend I was attending cos I'd much rather be
working on this and pretenders.

Expect much merging/commenting/pull requesting on both repos tomorrow...

File: SETUP.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Set up environment to run tests
+
+## Install dependencies
+
+* Make yourself a virtual environment, and activate it
+* Install python packages with `pip install -r requirements/test.txt`
+* Make yourself a node environment within the virtualenv with `nodeenv -p`
```
Comment (alexcouper) [inline SETUP.md:11]: Add a dependency on `curl` somewhere?

File: SETUP.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  (this installs node and npm and takes quite a while)
+* Install node packages with `./nodereqs.py`
+  (this installs the required files listed in `requirements/nodeenv.txt`)
+
+## Run the tests
+
+* Angular Unit tests: `scripts/angular-unit-test.sh`
```
Comment (alexcouper) [inline SETUP.md:18]: Also need Chrome as a dependency.

*installs chrome into virtual machine*

File: SETUP.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+* Angular E2E tests: `scripts/angular-e2e-test.sh` (requires a running server)
+* Python tests: `nosetests`
```
Comment (alexcouper) [inline SETUP.md:20]: All tests pass on mine!

File: scripts/pr-for-issue [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    echo "Usage: $0 issue github-username github-password"
+    exit -1
+fi
+
+issue=$1
+ghuser=$2
+pass=$3
+
+#hub pull-request -i $issue -b develop -h pretenders:story/$issue/$USER
+http --auth $ghuser:$pass POST https://api.github.com/repos/pretenders/deploystream/pulls issue=$issue head="pretenders:story/$issue/carles" base=develop
```
Comment (alexcouper) [inline scripts/pr-for-issue]: s/carles/$2?

File: deploystream/static/test/unit/controllersSpec.js [hunk 5]
```diff
@@ -5,16 +5,36 @@
 /* jasmine specs for controllers go here */
 
 describe('Feature list controller', function () {
-    var scope, ctrl;
+    var scope, ctrl, $httpBackend;
+
+    beforeEach(module('deploystream.services'));
+
+    beforeEach(inject(function (_$httpBackend_, $rootScope, $controller) {
```
Comment (alexcouper) [inline deploystream/static/test/unit/controllersSpec.js:12]: I guess I should read up about `beforeEach` and `inject`...

**Comment by alexcouper** [comment]
[out]

File: SETUP.md (hunk 1 above)
Comment (txels) [inline SETUP.md:11]: oooh cripes, hadn't realised `nodeenv` depends on `curl`. Which is quite a sneaky way to implement http downloads in a python tool...

**Comment by alexcouper** [comment]
(cant' be auto merged...muhahaha)

File: SETUP.md (hunk 2 above)
Comment (txels) [inline SETUP.md:18]: Yes, same as with `curl` this comes under os-level dependencies. Which got me thinking, maybe the best solution for this is to assume a VM setup and include a `Vagrantfile`. What do you think? It seemed to work well at UKTV where I also had people developing on ubuntus and macs.

Comment (alexcouper) [inline SETUP.md:18]: I'm OK with that :+1:

File: SETUP.md (hunk 3 above)
Comment (txels) [inline SETUP.md:20]: \o/

File: scripts/pr-for-issue (hunk 4 above)
Comment (txels) [inline scripts/pr-for-issue]: oops. Luckily I did not include my password.
`$2` would not work for me, as my username and github username are not the same, and I use `$USER` in the branch name (`start-issue` script), so it should probably be `$USER` in here (or `$2` in there). But if you want to use these scripts as well, we may use this opportunity to choose the naming convention we want to use in branches (e.g. `issue` vs `story`, etc.)

File: deploystream/static/test/unit/controllersSpec.js (hunk 5 above)
Comment (txels) [inline deploystream/static/test/unit/controllersSpec.js:12]: `beforeEach` is the easy one, `inject` is a tad more obscure.

**Comment by alexcouper** [comment]
I'm assuming that you're doing some vagrant-ing at the moment with this, so staying clear.

**Comment by txels** [comment]
Ha not really. I'm more playing with Bootstrap ATM

**Comment by alexcouper** [comment]
You got time for a quick Skype chat?

**Comment by txels** [comment]
I fixed the failing test in this branch.
I'm not sure it is worth doing the vagranting now, as it is still fairly simple to set the environment up, and both of us have it working now. I'd rather focus on other things. Feel free to merge this, we can add a story for vagrant if we feel the need later.

**Comment by alexcouper** [comment]
Fair point. Merging.

**Pull request merged**